core/modules/wdb/QueryRunner.cc

```diff
@@ -269,7 +273,11 @@ void QueryRunner::_transmitHeader(std::string& msg) {
     // Make sure protoheader size can be encoded in a byte.
     assert(protoHeaderString.size() < 255);
     auto msgBuf = proto::ProtoHeaderWrap::wrap(protoHeaderString);
-    _task->sendChannel->sendStream(msgBuf.data(), msgBuf.size(), false);
+    if (!_cancelled) {
```

Comment:
I don't know how this flag is set but it likely should have been set using an atomic operation. This appears to be just a regular old inspection without any locks. So, how do you interlock testing cancellation with the act of cancellation. I don't see it.